gprofiler/metadata/system_metadata.py

```diff
@@ -194,15 +205,39 @@ class SystemInfo:
     spawn_uptime_ms: float
 
 
+def get_windows_network_details() -> Tuple[str, str]:
+    IPV4_PATTERN = r"^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$"
+    LOOPBACK = "127.0.0.1"
+    ADDR_KEY = "addr"
+    MAC_KEY = -1000
+    for iface in netifaces.interfaces():
+        mac_address = None
+        for key, addresses in netifaces.ifaddresses(iface).items():
+            for address in addresses:
+                if key == MAC_KEY:
+                    mac_address = address[ADDR_KEY]
+                    continue
+                if re.match(IPV4_PATTERN, address[ADDR_KEY]) is not None and address[ADDR_KEY] != LOOPBACK:
```

Comment:
Instead of the homemade RE pattern for IPv4, you can use `ipaddress.IPv4Address` and see if it raises.

Reply:
Done